service_invocation/csharp/http/order-processor/Program.cs

```diff
@@ -1,17 +1,18 @@
-using System.Text.Json.Serialization;
-
 var builder = WebApplication.CreateBuilder(args);
 
 var app = builder.Build();
 
-if (app.Environment.IsDevelopment()) {app.UseDeveloperExceptionPage();}
+if (app.Environment.IsDevelopment())
+{
+    app.UseDeveloperExceptionPage();
+}
 
-app.MapPost("/orders", async context => {
-    var data = await context.Request.ReadFromJsonAsync<Order>();
-    Console.WriteLine("Order received : " + data);
-    await context.Response.WriteAsync(data.ToString());
+app.MapPost("/orders", (Order order) =>

```

Comment:
Thank you @davidfowl!.  This is a nice improvement when json properties map identically.


Q: can and should we show this Async?

Reply:
ping @davidfowl

Reply:
No need to do async here. That happens before the method is called with the order.